svthalia/Reaxit
# Refactor ListState and paginated screens

Part of #99.

### Summary
This replaces the old ListState in a way similar to what I did in #325. Introduces `PaginatedScrollView` and a `PaginatedCubit` abstract class to handle most logic for paginated lists/grids.

### How to test
Steps to test the changes you made:
1. Try out all of the paginated list screens (including search).
2. Check out their botton padding, which should now be consistent and include a `SafeArea`.
